simulationdataschema/model_system_review.py

```diff
@@ -0,0 +1,873 @@
+
+    symmetry = SubSection(sub_section=Symmetry.m_def, repeats=True)
+
+    chemical_formula = SubSection(sub_section=ChemicalFormula.m_def, repeats=False)
+
+    branch_label = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        Label of the specific branch in the hierarchical `ModelSystem` tree.
+        """,
+    )
+
+    branch_depth = Quantity(
+        type=np.int32,
+        description="""
+        Index refering to the depth of a branch in the hierarchical `ModelSystem` tree.
+        """,
+    )
+
+    # TODO add method to resolve labels and positions from the parent AtomicCell
+    atom_indices = Quantity(
+        type=np.int32,
+        shape=["*"],
+        description="""
+        Indices of the atoms in the child with respect to its parent. Example:
+            - We have SrTiO3, where `AtomicCell.labels = ['Sr', 'Ti', 'O', 'O', 'O']`. If
+            we create a `model_system` child for the `'Ti'` atom only, then in that child
+            `ModelSystem.model_system.atom_indices = [1]`. If now we want to refer both to
+            the `'Ti'` and the last `'O'` atoms, `ModelSystem.model_system.atom_indices = [1, 4]`.
+        """,
+    )
+
+    bond_list = Quantity(
+        type=np.int32,
+        # TODO improve description and add an example using the case in atom_indices
+        description="""
+        List of pairs of atom indices corresponding to bonds (e.g., as defined by a force field)
+        within this atoms_group.
+        """,
+    )
+
+    model_system = SubSection(sub_section=SectionProxy("ModelSystem"), repeats=True)
+
+    def resolve_system_type_and_dimensionality(
+        self, ase_atoms: ase.Atoms
+    ) -> Tuple[str, int]:
+        """
+        Determine the ModelSystem.type and ModelSystem.dimensionality using MatID classification analyzer:
+
+            - https://singroup.github.io/matid/tutorials/classification.html
+
+        Args:
+            ase.Atoms: The ASE Atoms structure to analyse.
+        Returns:
+            system_type (str): The system type as determined by MatID.
+            dimensionality (str): The system dimensionality as determined by MatID.
+        """
+        classification = None
+        system_type, dimensionality = self.type, self.dimensionality
+        if (
+            len(ase_atoms)
+            <= config.normalize.system_classification_with_clusters_threshold
+        ):
+            try:
+                classifier = Classifier(
+                    radii="covalent",
+                    cluster_threshold=config.normalize.cluster_threshold,
+                )
+                cls = classifier.classify(ase_atoms)
+            except Exception as e:
+                self.logger.warning(
+                    "MatID system classification failed.", exc_info=e, error=str(e)
+                )
+                return system_type, dimensionality
+
+            classification = type(cls)
```

Comment:
Just spotted this:

be careful with `type`, it's both very powerful and a bit of an unreliable tool. It holds a special place in Python (it's the only "real" class, etc.).
A very typical feedback from code checkers like Copilot would be to swap it with `isinstance`, which behaves more in line with what you want here. Anyhow, the best way is to check with a test.

Moreover, I'd avoid the use of the variable `cls`, since it is often used in a similar context as `self`.
A pydev's first instinct, when seeing it, is to check for `__new__()`. Just give it any other name (except `class` :p).